src/scene2d.rs

```diff
@@ -166,3 +159,89 @@ pub fn load_next_map(
         .loaded_maps
         .insert(game_state.current_map.clone());
 }
+
+
+// until bevy_tiled has better support for this, we have to reach in and create objects based on tiles
+// TODO: tile objects set up to clear on reload, but MapReadyEvent only runs first load, so they are not re-created
+pub fn create_tile_objects_system(
+    mut commands: Commands,
+    mut map_ready_events: EventReader<MapReadyEvent>,
+    query: Query<&MapContainer>,
+    maps: Res<Assets<Map>>,
+    transient_state: Res<TransientState>,
+    mut game_state: ResMut<Game>,
+    mut map_query: Query<(&mut CreatedMapEntities, &Handle<Map>)>,
+    config: Res<Config>,
+) {
+    for event in map_ready_events.iter() {
+        let map_entity = event.map_entity_option.expect("why didn't you give this map an entity?");
+        if let Ok(container) = query.get(map_entity) {
+            maps.get(container.asset.clone()).map(|map| {
+                let mut templates: HashMap<u32, Object> = Default::default();
+                // find all tiles with object layers
+                for tileset in map.map.tilesets.iter() {
+                    for tile in tileset.tiles.iter() {
+                        if let Some(group) = &tile.objectgroup {
+                            for obj in &group.objects {
+                                templates.insert(tileset.first_gid + tile.id, Object::new(&obj));
+                            }
+                        }
+                    }
+                }
+                // go through visibile layers for this map and add occlusion objects for tiles
+                // NOTE: for now this assumes the entire tile occludes
```

Comment:
Is this what you meant?
```suggestion
                // NOTE: for now this assumes the entire tile obstructs
```